BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/MainActivity.kt

```diff
@@ -0,0 +1,92 @@
+package com.example.basicmusicplayer
+
+import android.media.AudioManager
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import android.webkit.WebViewClient
+import android.media.MediaPlayer
+import android.media.AudioAttributes
+import android.provider.MediaStore.Audio
+import android.widget.ProgressBar
+import android.view.View
+import android.util.Log
+import android.widget.Button
+import android.widget.Toast
+import java.io.IOException
+
+//define main activity class and define properties
+class MainActivity : AppCompatActivity() {
+    private lateinit var loadingIndicator: ProgressBar;
+    private lateinit var btnPlayAudio : Button;
+    private lateinit var btnPauseAudio : Button;
```

Comment:
Could you combine the play and pause button. The player has three states—paused, playing, and loading—so the button should have three states:
1. Displays "Play" when the player is paused.
2. Displays "Pause" when the player is playing.
3. Displays a spinner and is disabled when loading.